src/PackageCompiler.jl

```diff
@@ -409,10 +410,21 @@ function create_sysimg_object_file(object_file::String,
         """
 
     # Make packages available in Main. It is unclear if this is the right thing to do.
-    for pkg in packages
-        print(julia_code_buffer, """
-            import $pkg
+    if import_into_main
+        for pkg in packages
+            print(julia_code_buffer, """
+                if isdefined(Main, Symbol("$pkg"))
+                    @warn(
```

Comment:
Did you check that users will see this warning?

Reply:
I saw it on a build once but let me confirm on a simple example